homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,254 @@
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_temperature",
+            "Outdoor Temperature",
+            "mdi:thermometer",
+            DEVICE_CLASS_TEMPERATURE,
+            hass.config.units.temperature_unit,
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.outdoorTemperature
+
+
+class LyricOutdoorHumiditySensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_outdoor_humidity",
+            "Outdoor Humidity",
+            "mdi:water-percent",
+            DEVICE_CLASS_HUMIDITY,
+            "%",
+        )
+
+    @property
+    def state(self) -> str:
+        """Return the state of the sensor."""
+        return self.device.displayedOutdoorHumidity
+
+
+class LyricNextPeriodSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_next_period_time",
+            "Next Period Time",
+            "mdi:clock",
+            DEVICE_CLASS_TIMESTAMP,
+        )
+
+    @property
+    def state(self) -> datetime:
+        """Return the state of the sensor."""
+        device: LyricDevice = self.device
+        time = dt_util.parse_time(device.changeableValues.nextPeriodTime)
+        now = dt_util.utcnow()
+        if time <= now.time():
+            now = now + timedelta(days=1)
+        return dt_util.as_utc(datetime.combine(now.date(), time))
+
+
+class LyricSetpointStatusSensor(LyricSensor):
+    """Defines a Honeywell Lyric sensor."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+    ) -> None:
+        """Initialize Honeywell Lyric sensor."""
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_setpoint_status",
+            "Setpoint Status",
+            "mdi:thermostat",
```

Comment:
Icon is 👍  here since there is no DEVICE_CLASS, would be cleaner to pass `None` for the device class in case this gets refactored later